### ISSUE TITLE
Incorrect vocalization of the accessible name of the export modal

### ISSUE THREAD

**Issue by cyberbaloo** [suitenumerique/docs#2011]
## Bug Report

**Problematic behavior**
The accessible name of the modal that is rendered is [object Object], which is not relevant.

**Expected behavior/code**
The accessible name of the modal should be "Export document". Perhaps replace the aria-label attribute with the aria-labelledby="modal-export-title" attribute which would reference the <h1 id="modal-export-title">Export document</h1> already present.

**Steps to Reproduce**
1. Log in to a document in read mode
2. Enable the screen reader
3. Navigate into the share modale
4. And then the bug happens!

**Critères RGAA**

**Critère 7.1 :** [Chaque script est-il, si nécessaire, compatible avec les technologies d’assistance ?](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#7.1)

**Impact** 

The impact is blocking for screen readers.

The user may be confused by the incorrect vocalization of the modal's accessible name when it opens. It is incomprehensible and does not allow them to know what context they are in.

**Comment by FocusMode-coder** [comment]
Hey! I may be able to help with this bug.

I build custom fixes for Python projects and automation systems.

If you'd like help resolving this issue quickly feel free to reach out.

Happy to take a look 👍
